Tests/UnitTests/Misc/SandboxEnvironmentDetectorTests.swift

```diff
@@ -16,46 +16,96 @@ import XCTest
 
 @testable import RevenueCat
 
+final class MockLocalReceiptFetcher: LocalReceiptFetcherType {
+
+    let mockReceipt: AppleReceipt
+    let failReceiptParsing: Bool
+
+    init(mockReceipt: AppleReceipt, failReceiptParsing: Bool) {
+        self.mockReceipt = mockReceipt
+        self.failReceiptParsing = failReceiptParsing
+    }
+
+    func fetchAndParseLocalReceipt() throws -> RevenueCat.AppleReceipt {
+        if failReceiptParsing {
+            throw PurchasesReceiptParser.Error.receiptParsingError
+        }
+        return mockReceipt
+    }
+}
+
 class SandboxEnvironmentDetectorTests: TestCase {
 
-    func testIsSandbox() {
-        expect(SystemInfo.with(receiptResult: .sandboxReceipt, inSimulator: false).isSandbox) == true
+    func testIsSandboxIfReceiptURLIsSandbox() {
+        expect(SystemInfo.with(receiptURLResult: .sandboxReceipt).isSandbox) == true
     }
 
-    func testIsNotSandbox() {
-        expect(SystemInfo.with(receiptResult: .receiptWithData, inSimulator: false).isSandbox) == false
+    func testIsNotSandboxIfReceiptURLIsAppStore() {
+        expect(SystemInfo.with(receiptURLResult: .appStoreReceipt).isSandbox) == false
     }
 
     func testIsNotSandboxIfNoReceiptURL() {
-        expect(SystemInfo.with(receiptResult: .nilURL, inSimulator: false).isSandbox) == false
+        expect(SystemInfo.with(receiptURLResult: .nilURL).isSandbox) == false
     }
 
-    func testMacSandboxReceiptIsSandbox() {
-        expect(SystemInfo.with(receiptResult: .macOSSandboxReceipt, inSimulator: false).isSandbox) == true
+    func testIsAlwaysSandboxIfRunningInSimulator() {
+        expect(SystemInfo.with(receiptURLResult: .sandboxReceipt, inSimulator: true).isSandbox) == true
+        expect(SystemInfo.with(receiptURLResult: .appStoreReceipt, inSimulator: true).isSandbox) == true
+        expect(SystemInfo.with(receiptURLResult: .nilURL, inSimulator: true).isSandbox) == true
     }
 
-    func testMacAppStoreReceiptIsNotSandbox() {
-        expect(SystemInfo.with(receiptResult: .macOSAppStoreReceipt, inSimulator: false).isSandbox) == false
+    func testIsNotSandboxIfReceiptIsProduction() throws {
+        try AvailabilityChecks.skipIfNotMacOS()
+
+        expect(SystemInfo.with(receiptEnvironment: .production).isSandbox) == false
     }
 
-    func testIsAlwaysSandboxIfRunningInSimulator() {
-        expect(SystemInfo.with(receiptResult: .sandboxReceipt, inSimulator: true).isSandbox) == true
-        expect(SystemInfo.with(receiptResult: .receiptWithData, inSimulator: true).isSandbox) == true
-        expect(SystemInfo.with(receiptResult: .nilURL, inSimulator: true).isSandbox) == true
+    func testIsSandboxIfReceiptIsNotProduction() throws {
+        try AvailabilityChecks.skipIfNotMacOS()
+
+        expect(SystemInfo.with(receiptEnvironment: .sandbox).isSandbox) == true
+    }
+
+    func testIsSandboxIfReceiptParsingFailsAndBundleSignatureIsNotMacAppStore() throws {
+        try AvailabilityChecks.skipIfNotMacOS()
```

Comment:
Added some tests for macOS, which do not currently run so I cannot properly test them. Once the macOS tests PR is ready I'll make sure to fix any errors if tests don't pass.